ubuntu-advantage

```diff
@@ -313,10 +325,50 @@ is_livepatch_enabled() {
     canonical-livepatch status >/dev/null 2>&1
 }
 
-validate_user_pass_token(){
+validate_user_pass_token() {
     echo "$1" | grep -q '^[^:]\+:[^:]\+$'
 }
 
+check_url_with_credentials() {
+    local repo_url="$1"
+    local credentials="$2"
+    local file="$3"
+
+    local url
+    url=$(private_repo_url "$repo_url" "$credentials" "$file")
+
+    local tempfile log
+    tempfile="$(mktemp)"
+    log="$(mktemp)"
+    if "$APT_HELPER" download-file "$url" "$tempfile" >"$log" 2>&1; then
+        rm -f "$tempfile" "$log"
+        return 0
+    fi
+
+    local error_line
+    error_line=$(sed -n 's/^E: Failed to fetch [^ ]\+ \+//p' "$log")
+    if echo "$error_line" | grep -q '^\(HttpError\)\?401'; then
+        error_msg 'Invalid credentials'
+    else
+        error_msg "Failed checking credentials ($error_line)"
+    fi
+
+    rm -f "$tempfile" "$log"
+    return 1
+}
+
+check_credentials() {
+    local repo_url="$1"
+    local credentials="$2"
+
+    # skip the check if the helper is not found
+    [ -x "$APT_HELPER" ] || return 0
+
+    echo -n 'Checking credentials... '
```

Comment:
What do you think about doing the APT_HELPER check after this echo, and output "SKIP" or "SKIPPED (no apt-helper found)", or some combination of these, if we don't find it?

Reply:
Good point, done